projects/igniteui-angular/src/lib/directives/date-time-editor/date-time-editor.directive.ts

```diff
@@ -135,7 +136,17 @@ export class IgxDateTimeEditorDirective extends IgxMaskDirective implements OnCh
      * ```
      */
     @Input()
-    public displayFormat: string;
+    public set displayFormat(value: string) {
+        if (value) {
+            this._displayFormat = value;
+        }
+        this.updateDefaultFormat();
+        this.setMask(this.inputFormat);
```

Comment:
Same here for the value check. Also if this set mask popped here, wouldn't there be an equivalent place it moved from?